pennylane/math/quantum.py

```diff
@@ -429,6 +429,10 @@ def reduce_statevector(state, indices, check_state=False, c_dtype="complex128"):
     else:
         batch_dim, dim = np.shape(state)[:2]
 
+        # batch dim exists but is unknown; cast to int so that reshaping works
+        if batch_dim is None:

```

Comment:
Is it specific for tf autograph?

Reply:
Yes, because during autograph compilation the batch dimension is unknown, e.g. the tensor has shape `(None, 2**n)` here.